access/views.py

```diff
@@ -481,6 +507,7 @@ def test_result(request):
     return HttpResponse(result or 'No test result received yet.')
 
 
+@instance_read_access_required
```

Comment:
The generated exercise files are supposed to be public to the web so that the students may download them. The download link in the A+ page points directly to this mooc-grader endpoint. So the login decorator should be removed, right?

As an example, see the generated file links inside these personalized exercises:
https://plus.cs.aalto.fi/aplus-manual/master/programming_exercises/personalized_exercises/